core/src/main/java/org/elasticsearch/common/settings/SecureSetting.java

```diff
@@ -46,6 +46,7 @@
     private SecureSetting(String key, Property... properties) {
         super(key, (String)null, null, ArrayUtils.concat(properties, FIXED_PROPERTIES, Property.class));
```

Comment:
How about `super(KeyStoreWrapper.validateSettingName(key), ...)` and make `validateSettingName` return `key` if it passes validation?

Reply:
I don't think that buys anything here, since the object will not be fully constructed until the ctor exits anyways. And it makes the semantics of the validation method a little weird (eg could it mean the return value *must* be used?). I'm going to keep as is but happy to discuss and modify later if you insist.

Reply:
Yes, I am fine with that.